dataland-frontend/src/utils/DataTableDisplay.ts

```diff
@@ -43,3 +43,83 @@ export type ReportingPeriodOfDataSetWithId = {
   dataId: string;
   reportingPeriod: string;
 };
+
+/**
+ * Adds click event listeners on DataTable row headers to expand and collapse row
+ * @param expandedRowsOnClick function passes the latest list of expanded row id's
+ * @param newExpandedRowsCallback function that returns the updated (after click) list of expanded row id's
+ * @returns the map of rows and their click handlers needed for unmounting
+ */
+export function mountRowHeaderClickEventListeners(
+  expandedRowsOnClick: () => string[],
```

Comment:
Why isn't this simply inserted as a list of strings?  I don't see the need of a function as input which returns the list.

Reply:
Each clickHandler has it's own scope and do not know whether other clicks were made by other rows... So every time we click a row, we request the latest list of expanded rows in order to add or remove the currently clicked item... and send the update dlist back.

Reply:
Yes, but anyways, the `mountRowHeaderClickEventListeners` function could receive the latest list of expanded rows directly as list of strings.
Currently it receives a function, which returns the list.   I don't see the need for that function.   Why not directly passing the list to it?:

`export function mountRowHeaderClickEventListeners(
  expandedRowsOnClick: string[],
  newExpandedRowsCallback: (newExpandedRows: string[]) => void,
) .....`

Reply:
That way that function only gets the list once... on mount only.
We need the value every time we click. So onclick we call that function which returns the newest, up-to-date list.

Reply:
Ahh ok got it! Thx for the explanation 👍